sdks/python/apache_beam/io/mongodbio.py

```diff
@@ -460,16 +457,15 @@ def _get_split_keys(
     if self._range_is_not_splittable(start_pos, end_pos):
       return []
 
-    with MongoClient(self.uri, **self.spec) as client:
```

Comment:
this changed timing of client, made if from transient and auto-closing to persistent.

I would prefer a fix not changing current behavior. Consider the following

1. Revert #36949
2. In `__init__`, update self.spec such that, if it does not contain a key named "driver", set `self.spec["driver"] = DriverInfo("Apache Beam", beam.__version__)`